nncf/torch/quantization/init_range.py

```diff
@@ -122,15 +133,19 @@ def generate_collectors_for_range_init_statistics_collection(target_model_graph:
     @staticmethod
     def generate_stat_collector_for_range_init_config(
             init_config: RangeInitConfig,
+            is_weights: bool,
+            rs_vs_params: Dict[ReductionShape, Dict[str, Union[str, bool]]] = None,
             reduction_shapes: Set[ReductionShape] = None,
             num_samples_to_collect_override: int = None) -> TensorStatisticCollectorBase:
         num_samples = init_config.num_init_samples
         if num_samples_to_collect_override is not None:
             num_samples = num_samples_to_collect_override
         if init_config.init_type == "min_max":
-            return MinMaxStatisticCollector(reduction_shapes, num_samples)
+            return MinMaxStatisticCollector(reduction_shapes, rs_vs_params, num_samples)
+        if init_config.init_type == "mixed_min_max":
+            return MixedMinMaxStatisticCollector(is_weights, reduction_shapes, rs_vs_params, num_samples)
         if init_config.init_type == "mean_min_max":
-            return MeanMinMaxStatisticCollector(reduction_shapes, num_samples)
+            return MeanMinMaxStatisticCollector(reduction_shapes, rs_vs_params, num_samples)
```

Comment:
`reduction_shapes` is not really needed here. You could iterate over values of `rs_vs_params` dictionary to get the same.
the redundancy can be solved by passing more general `rs_vs_params` for the base class.

Reply:
or maybe by having a different constructor with params that are actually needed...

Reply:
Thanks for the comment. `reduction_shapes` are removed. Now all collectors accept only `rs_vs_params` and it is processed further accordingly. 
Only `min_max` based collectors require some parameters other than `reduction_shapes`. That is why I propose to extract the `reduction_shapes` and pass it to the base class, keeping other params at the level of the collectors.